tests/ml_tools/test_train.py

```diff
@@ -200,3 +207,122 @@ def test_fit_sin_adjoint(BasicAdjointQNN: torch.nn.Module) -> None:
 
     x_test = torch.rand(1, 1)
     assert torch.allclose(torch.sin(x_test), model(x_test), rtol=1e-1, atol=1e-1)
+
+
+def test_train_dataloader_raisesError_if_val_check_is_True_but_non_dict_dataloader(
```

Comment:
Could you try to simplify these tests a little bit? There's some code duplication in all three of them, i.e. the `TrainConfig` seems to be the same, so maybe you can separate that into a function that is reused. 

Also no need for the super long test titles :) If you wish to have some extra context to what the test is doing that's cool of course, but best to write in a docstring or comment just below I think.

Reply:
Thanks.

> There's some code duplication in all three of them, i.e. the TrainConfig seems to be the same, so maybe you can separate that into a function that is reused.

I agree that it helps reduce space, so I have factored out config creation.

> Also no need for the super long test titles :) If you wish to have some extra context to what the test is doing that's cool of course, but best to write in a docstring or comment just below I think.

Okay, great. I have reduced the lengths of the test titles. Hopefully, that makes it cleaner!